scripts/cmake/vcpkg_host_path_list.cmake

```
@@ -42,20 +22,36 @@ function(vcpkg_host_path_list)
     set(list_var "${ARGV1}")
 
     cmake_parse_arguments(PARSE_ARGV 2 arg "" "" "")
-    z_vcpkg_translate_to_host_path_list(arguments "${arg_UNPARSED_ARGUMENTS}")
+    set(bad_items "${arg_UNPARSED_ARGUMENTS}")
+    list(FILTER bad_items INCLUDE REGEX "[${VCPKG_HOST_PATH_SEPARATOR}]")
+    if(NOT "${bad_items}" STREQUAL "")
+        message(FATAL_ERROR "Host path separator (${VCPKG_HOST_PATH_SEPARATOR}) in path; this is unsupported.")
+    endif()
 
     if("${operation}" STREQUAL "SET")
+        cmake_path(CONVERT "${arg_UNPARSED_ARGUMENTS}" TO_NATIVE_PATH_LIST arguments)
         set(list "${arguments}")
-    elseif("${operation}" MATCHES "^(APPEND|PREPEND)$")
-        if("${arguments}" STREQUAL "")
-            # do nothing
-        elseif("${list}" STREQUAL "")
+    elseif("${operation}" STREQUAL "APPEND")
+        cmake_path(CONVERT "${arg_UNPARSED_ARGUMENTS}" TO_NATIVE_PATH_LIST arguments)
+        if("${list}" STREQUAL "")
             set(list "${arguments}")
-        elseif("${operation}" STREQUAL "PREPEND")
-            set(list "${arguments}${VCPKG_HOST_PATH_SEPARATOR}${list}")
-        else()
+        elseif(NOT "${arguments}" STREQUAL "")
             set(list "${list}${VCPKG_HOST_PATH_SEPARATOR}${arguments}")
         endif()
+    elseif("${operation}" STREQUAL "PREPEND")
+        cmake_path(CONVERT "${arg_UNPARSED_ARGUMENTS}" TO_NATIVE_PATH_LIST arguments)
+        if("${list}" STREQUAL "")
+            set(list "${arguments}")
+        elseif(NOT "${arguments}" STREQUAL "")
+            set(list "${arguments}${VCPKG_HOST_PATH_SEPARATOR}${list}")
+        endif()
+    elseif("${operation}" STREQUAL "REMOVE_DUPLICATES")
+        if(NOT VCPKG_HOST_PATH_SEPARATOR STREQUAL ";")
+            string(REPLACE [[;]] [[\\\;]] list "${list}")

```

```suggestion
            string(REPLACE [[;]] [[\;]] list "${list}")
```

Doesn't `\;` already escape it in the list? `\\\` looks like it was meant for `""`'d strings rather than `[]` ones?

Or am I just confused?

---

This line prepares the input for 
~~~
cmake_path(CONVERT "${list}" TO_CMAKE_PATH_LIST current_list)
~~~
At this point, `list` is a native host past list, not a cmake list.
(Windows `ENV{PATH}`, I'm looking at you.)

There are tests for individual platforms, guarded by the value of `HOST_PATH_SEPARATOR`.